docs/pages/kubernetes-access/discovery/google-cloud.mdx

```diff
@@ -0,0 +1,578 @@
+```
+
+Attach your service account to the VM running the Discovery Service:
+
+```code
+$ gcloud compute instances set-service-account ${VM2_NAME?} \
+   --service-account teleport-discovery-service@${PROJECT_ID?}.iam.gserviceaccount.com \
+   --zone ${MY_ZONE?} \
+   --scopes=cloud-platform
+```
+
+</Details>
+
+<Notice type="warning">
+
+You must use the `scopes` flag in the `gcloud compute instances
+set-service-account` command. Otherwise, your Google Cloud VM will fail to
+obtain the required authorization to access the GKE API.
+
+</Notice>
+
+Once you have attached the service account, restart your VM:
+
+```code
+$ gcloud compute instances start ${VM_NAME?} --zone ${MY_ZONE?}
+```
+</TabItem>
+<TabItem label="Other Platform">
+
+Download a credentials file for the service account used by the Discovery
+Service and Kubernetes Service:
+
+```code
+$ PROJECT_ID=project-id
+$ gcloud iam service-accounts keys create google-cloud-credentials.json \
+    --iam-account=teleport-discovery-kubernetes@${PROJECT_ID?}.iam.gserviceaccount.com
+```
+
+Move your credentials file to the host running the Teleport Discovery Service
+and Kubernetes Service the path
+`/var/lib/teleport/google-cloud-credentials.json`. We will use this credentials
+file when running this service later in this guide.
+
+<Details title="Deploying the Kubernetes Service and Discovery Service
+separately?">
+
+Download separate credentials files for each service:
+
+```code
+$ PROJECT_ID=project-id
+$ gcloud iam service-accounts keys create discovery-service-credentials.json \
+    --iam-account=teleport-discovery-service@${PROJECT_ID?}.iam.gserviceaccount.com
+$ gcloud iam service-accounts keys create kube-service-credentials.json \
+    --iam-account=teleport-kubernetes-service@${PROJECT_ID?}.iam.gserviceaccount.com
+```
+
+Move `discovery-service-credentials.json` to the host running the Teleport
+Discovery Service at the path `/var/lib/teleport/google-cloud-credentials.json`. 
+
+Move `kubernetes-service-credentials.json` to the host running the Teleport
+Kubernetes Service at the path
+`/var/lib/teleport/google-cloud-credentials.json`. 
+
+We will use these credentials files when running this services later in this
+guide.
+
+</Details>
+
+</TabItem>
+</Tabs>
+
+## Step 2/3. Configure Teleport to discover GKE clusters
+
+Now that you have created a service account that can discover GKE clusters and a
+cluster role that can manage access, configure the Teleport Discovery Service to
+detect GKE clusters and the Kubernetes Service to proxy user traffic.
+
+### Install Teleport
+
+Install Teleport on the host you are using to run the Kubernetes Service and
+Discovery Service:
+
+(!docs/pages/includes/install-linux.mdx!)
+
+### Create a join token 
+
+The Teleport Discovery Service and Kubernetes Service require an authentication
+token in order to to join the cluster. Generate one by running the following
+`tctl` command:
+
+```code
+$ tctl tokens add --type=discovery,kube
```

Comment:
For those running Discovery and Kubernetes services separately, does this command/section need another version in a Details block, split out into tokens for each service